### PR TITLE
Don't repeat already posted warnings

### LINKED ISSUE #3: Avoid repeating comments
The tool currently does not care whether an identical comment already exists on the same line of code.
Check if there is already a comment before posting a new one.

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
Hi @platisd I would like to suggest another minor improvement that should prevent extra warnings from being spammed and close #3.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,10 +29,7 @@ Please note the following:
   comments on other files outside the pull request `diff`.
   This means that there may be more warnings in your project. Make sure you fix
   them *before* starting to use this Action to ensure new warnings will not be introduced in the future.
-* Having the Action leave comments on every update of the pull request may result in spam. Specifically,
-  *existing* comments are **not** taken into consideration and will be *repeated* every time the Action is triggered,
-  as long as the violation has not been amended.
-  If you see this becoming an issue, consider triggering it manually with a comment instead (see example below).
+* This Action *respects* existing comments and doesn't repeat the same warnings for the same line (no spam).
```
Comment (platisd) [inline README.md:32]: :tada: :pray:

File: run_action.py [hunk 2]
```diff
@@ -229,6 +229,55 @@ def main():
...
+                + str(pull_request_comments_result.status_code)
+            )
+            return 1
+
+        pull_request_comments = json.loads(pull_request_comments_result.text)
+
+        if len(pull_request_comments) == 0 :
+            break
+
+        existing_pull_request_comments = existing_pull_request_comments + pull_request_comments
```
Comment (platisd) [inline run_action.py:262]: Minor comment for a future pull request or... never: Can become `existing_pull_request_comments += pull_request_comments`

File: run_action.py [hunk 3]
```diff
@@ -229,6 +229,55 @@ def main():
...
+    for comment in existing_pull_request_comments:
+        review_comments = list(
+            filter(
+                lambda review_comment: not (
+                    review_comment["path"] == comment["path"] and
+                    review_comment["line"] == comment["line"] and
+                    review_comment["body"] == comment["body"]
+                ),
+                review_comments,
+            )
```
Comment (platisd) [inline run_action.py:274]: Nice :+1:

**Review by platisd — APPROVED** [review]
Great stuff! Thanks for contributing and solving the issue! :t-rex:

File: run_action.py (hunk 2 above)
Comment (oleg-derevenetz) [reply run_action.py:262]: Sorry for this. You don't always notice such things right away :)

**Pull request merged**